lib/client/client.js

```diff
@@ -1273,11 +1273,18 @@ Client.prototype._sendSocket = function _sendSocket (message,
   log.trace('sending request %j', message.json)
 
   try {
-    return conn.write(message.toBer(), writeCallback)
+    if (expect === 'unbind') {
+      return conn.write(message.toBer(), writeCallback)
+    } else {
+      writeCallback()
+      return conn.write(message.toBer())
+    }
```

Comment:
At a minimum we should have a comment here that explains why this code is the way it is. This would at least allow us to accept it without a test, as I know there is difficult code in this project. However, we really should be able to force the condition in our unit tests somehow.